src/authClient.js

```diff
@@ -23,17 +29,44 @@ export default (client, options = {}) => (type, params) => {
         password,
       });
     case AUTH_LOGOUT:
+      localStorage.removeItem(permissionsKey);
       return client.logout();
     case AUTH_CHECK:
       return localStorage.getItem(storageKey) ? Promise.resolve() : Promise.reject();
     case AUTH_ERROR:
-      const { code } = params;
+      const { code } = params
       if (code === 401) {
         localStorage.removeItem(storageKey);
+        localStorage.removeItem(permissionsKey);
         return Promise.reject();
       }
-      return Promise.resolve();
+      return Promise.resolve()
+    case AUTH_GET_PERMISSIONS:
+      /*
+      JWT token may be providen by oauth, 
+      so that's why the permissions are decoded here and not in AUTH_LOGIN.
+      */
+      //Get the permissions from localstorage if any.
+      const permissions = JSON.parse(localStorage.getItem(permissionsKey));
```

Comment:
localStorage can be manipulated. See https://stackoverflow.com/questions/9404813/how-to-view-or-edit-localstorage#comment78227724_12799050

Reply:
Yes it is true, here we are using permissions to adapt the display to user's permissions. You have to manage permissions too on the api. Here is a sample hook for that:
```javascript
restrictToRoles({
    roles: [
      'admin',
    ],
    fieldName: 'roles',
  })
```